packages/settingeditor/src/SettingsFormEditor.tsx

```diff
@@ -332,10 +332,11 @@ export class SettingsFormEditor extends React.Component<
   SettingsFormEditor.IProps,
   SettingsFormEditor.IState
 > {
+  private formState: any;
   constructor(props: SettingsFormEditor.IProps) {
     super(props);
     const { settings } = props;
-    this.state = {
+    this.formState = {
```

Comment:
The formData pass to `Form` is indeed only the [initial value](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/form-props/#formdata). So it makes sense to extract `formData` from the state. However `isModified` should be kept in the state otherwise the _Restore to Defaults_ button does not appear when needed.

Reply:
Putting back `isModified` in the react state generate rendering and looses the focus (the **Restore to Defaults** button is show again).

I have tried alternative approaches https://github.com/jupyterlab/jupyterlab/issues/12160#issuecomment-1108897013 without success.

Reply:
I opened https://github.com/datalayer-externals/jupyterlab/pull/3 with a better approach that only update props appropriately avoiding unnecessary component rerendering that were creating the focus lost.

Reply:
Awesome. I have merged your changes. Thx a bunch for this.